fedimint-core/src/db/mod.rs

```diff
@@ -286,6 +313,79 @@ pub trait IDatabaseTransaction<'a>: 'a + Send {
     async fn set_tx_savepoint(&mut self);
 }
 
+pub struct IDatabaseTransactionWrapper<'a, Tx: IDatabaseTransactionCore<'a> + Send>(
+    Option<Tx>,
```

Comment:
`Wrapper` also not very informative.

And it shouldn't be named `*I*Database...` as it is no longer an interface/trait?

`SingleUseDatabaseTransaction<Tx>` maybe?

Reply:
Ugh yeah, was struggling to come up with the name. I like `SingleUseDatabaseTransaction`. Here's what I renamed everything to:

`IDatabaseTransactionCore` -> `IDatabaseTransaction`
`IDatabaseTransaction` -> `ISingleUseDatabaseTransaction`
`IDatabaseTransactionWrapper` -> `SingleUseDatabaseTransaction`

Hopefully that's a bit more clear.